File: 2024/day02/day02_test.go
package day02

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestPart1(t *testing.T) {
	assert.Equal(t, 2, Part1("input_example_1"))
	assert.Equal(t, 585, Part1("input"))
}

// func TestPart2(t *testing.T) {
// 	assert.Equal(t, 31, Part2("input_example_1"))
// 	assert.Equal(t, 21306195, Part2("input"))
// }

File: 2025/test/day_02_test.exs
defmodule Day02Test do
  use ExUnit.Case

  import TestHelper

  test "part1" do
    assert Day02.part_1(input("day_02_example_1")) == 1_227_775_554
    assert Day02.part_1(input("day_02")) == 12_599_655_151
  end

  test "part2" do
    assert Day02.part_2(input("day_02_example_1")) == 4_174_379_265
    assert Day02.part_2(input("day_02")) == 20_942_028_255
  end
end
